lib_eio_posix/eio_posix_stubs.c

```diff
@@ -161,6 +163,68 @@ CAMLprim value caml_eio_posix_mkdirat(value v_fd, value v_path, value v_perm) {
   CAMLreturn(Val_unit);
 }
 
+static double double_of_timespec(struct timespec *t) {
+  return ((double) t->tv_sec) + (((double ) t->tv_nsec) / 1e9);
+}
+
+static value get_file_type_variant(struct stat sb) {
+  int filetype = sb.st_mode & S_IFMT;
+  if (filetype == S_IFREG) {
+    return caml_hash_variant("Regular_file");
+  } else if (filetype == S_IFSOCK) {
+    return caml_hash_variant("Socket");
+  } else if (filetype == S_IFLNK) {
+    return caml_hash_variant("Symbolic_link");
+  } else if (filetype == S_IFBLK) {
+    return caml_hash_variant("Block_device");
+  } else if (filetype == S_IFDIR) {
+    return caml_hash_variant("Directory");
+  } else if (filetype == S_IFCHR) {
+    return caml_hash_variant("Character_special");
+  } else if (filetype == S_IFIFO) {
+    return caml_hash_variant("Fifo");
+  } else {
+    return caml_hash_variant("Unknown");
+  }
+}
+
+CAMLprim value caml_eio_posix_fstatat(value v_fd, value v_path, value v_flags) {
+  CAMLparam1(v_path);
+  CAMLlocal1(v_ret);
+  char *path;
+  int ret;
+  struct stat statbuf;
+
+  caml_unix_check_path(v_path, "fstatat");
+  path = caml_stat_strdup(String_val(v_path));
+  caml_enter_blocking_section();
+  ret = fstatat(Int_val(v_fd), path, &statbuf, Int_val(v_flags));
+  caml_leave_blocking_section();
+  caml_stat_free_preserving_errno(path);
+  if (ret == -1) uerror("fstatat", v_path);
+
+  v_ret = caml_alloc_small(12, 0);
```

Comment:
```suggestion
  v_ret = caml_alloc(12, 0);
```
This ensures the new record is populated with GC-safe values (I already fixed this in the uring PR).